### ISSUE TITLE
Not getting consistent results with .h5 and .tflite models on different machines

### ISSUE THREAD

**Issue by rohanmishra21** [tensorflow/tensorflow#53908]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Tested on Windows 11, Intel Mac, Mac M1 v11.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): binary (tensorflow for windows and tensorflow-macos for Mac)
- TensorFlow version (use command below): 2.5.0
- Python version: 3.8.12
- Bazel version (if compiling from source): No
- GCC/Compiler version (if compiling from source): No
- CUDA/cuDNN version: 11.0 (for training only)
- GPU model and memory: Mac M1

**Describe the current behavior**
The same h5 and/or TFLite model gives different outputs on different machines. For an image regression model where ground truth labels range between [0, 3] the results differ by ~0.2-0.3 on different machines, which causes thresholding issues.

**Describe the expected behavior**
The same model should theoretically perform the same on all machines or at least they should match upto more precision.

**Standalone code to reproduce the issue**
The current model arch and the compilation info:
```
base_model = tf.keras.applications.MobileNetV3Small(input_shape=(IMAGE_SIZE, IMAGE_SIZE, 3),
                                               include_top=False,
                                               weights='imagenet',
                                               minimalistic=True)

base_model.trainable = True
    
model = tf.keras.Sequential([
        base_model,
        #pretrained_model,
        tf.keras.layers.GlobalAveragePooling2D(),
        #tf.keras.layers.Dense(256,activation = 'relu',dtype=tf.float32),
        tf.keras.layers.Dense(2, dtype=tf.float32)
    ])

model.compile(
        loss = tf.keras.losses.MeanAbsoluteError(),
        optimizer = tf.keras.optimizers.Adam(),
        metrics = [tf.keras.metrics.MeanAbsoluteError(), tf.keras.metrics.MeanSquaredError()],
        steps_per_execution=64
    )

model.summary()
```
```
Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
MobilenetV3small (Functional (None, 4, 4, 1024)        1031848   
_________________________________________________________________
global_average_pooling2d (Gl (None, 1024)              0         
_________________________________________________________________
dense (Dense)                (None, 2)                 2050      
=================================================================
Total params: 1,033,898
Trainable params: 1,021,786
Non-trainable params: 12,112
```
The model was converted to the usual TFLite format as well, but the issue persisted.
```
# Convert the model.
converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()
```

The model, thus obtained, can be used to reproduce the issue on different machines. Any ideas on why it might be happening and how to deal with it would mean a lot. Thanks!

**Comment by sushreebarsa** [comment]
@rohanmishra21 Could you please try on the latest version of TF v2.7.0 and let us know the outcome?Thanks!

**Comment by rohanmishra21** [comment]
@sushreebarsa Thanks for the reply! It seemed to be some issue with the tensorflow-metal plugin on the mac which led to inconsistencies, removing it solved the issue. Closing this issue now :)
